crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -123,12 +129,25 @@ impl<'a> ObjectRuntime<'a> {
         ty: Type,
         tag: StructTag,
         obj: Value,
-    ) -> PartialVMResult<()> {
+    ) -> PartialVMResult<TransferResult> {
         let id: ObjectID = get_object_id(obj.copy_value()?)?
             .value_as::<AccountAddress>()?
             .into();
+        let transfer_result =
+            if self.state.new_ids.contains_key(&id) || id == SUI_SYSTEM_STATE_OBJECT_ID {
```

Comment:
Second part of this conditional is kind of surprising--is this a special case for genesis? Might be worth mentioning if so.

Reply:
Yeah worth a comment. The system state object is special in that its ID is not generated from TxContext, but hard coded at 0x5. It maybe clearer if we could find a cleaner way to augment new_ids to include it in the genesis case.